lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -482,6 +476,96 @@ def operations
       end
     end # End of POLICY_ELEMENT_TYPES iteration
 
+    # Given a policy element class and a set of conditions, returns an
+    # ActiveRecord_Relation with those conditions applied
+    def build_active_record_relation(pe_class:, conditions:, ignore_case:)
+      # If any condition is case-insensitive, the nodes need to be built
+      # individually with Arel.
+      if ignore_case
+        conditions.map do |k, v|
+          # Arel matches provides agnostic case insensitive sql for MySQL and
+          # Postgres. This should always evaluate to an ActiveRecord_Relation.
+          if ignore_case_applies?(ignore_case, k)
+            build_arel_insensitive(pe_class: pe_class, key: k, value: v)
+          else
+            build_arel_sensitive(pe_class: pe_class, key: k, value: v)
+          end
+        # Reduce Arel nodes into ActiveRecord_Relation
+        end.reduce(pe_class.where(nil)) { |rel, e| rel.where(e) }
+      else
+        # If all conditions are case-sensitive, a direct where call can be used.
+        pe_class.where(conditions.to_h)
+      end
+    end
+
+    # Build Arel nodes using case-insensitive matching
+    def build_arel_insensitive(pe_class:, key:, value:)
+      unless value.is_a?(Array)
```

Comment:
https://github.com/rubocop-hq/ruby-style-guide#no-else-with-unless

Reply:
Could also make this method
```ruby
def build_arel_insensitive(pe_class:, key:, value:)
  pe_class.arel_table[key].matches_any([value].flatten(1))
end
```
...not sure if that's clearer, though.

Reply:
That's a tricky call. I think I would prefer being explicit about what we're expecting `value` to possibly be... I think it would be better to err on the side of being verbose over concise (just for straightforwardness) in these two methods, but I'm not 100% attached to that.

Reply:
That works for me 👍